### ISSUE TITLE
Separate out the Form BC

### ISSUE THREAD

**Issue by stevelacey** [LExpress/symfony1#54]
A lot of the embedded form stuff is broken by the non-BC Form changes, unsurprisingly, but if you're using a lot of this (as well as plugins that use it), and just need to maintain support for the project, it's hardly worth rewriting it just to maintain compatibility - is it worth separating this out from the other efforts?

**Comment by boutell** [comment]
Fourthed - we've been closely evaluating this fork this week with an eye to migrating our legacy projects to it as a maintained version of Symfony 1.4, but we do need backwards compatibility, so it looks like we can't do it at this stage.

**Comment by boutell** [comment]
I've posted on the Symfony 1 google group asking if there is interest in a full-bc, bare-minimum-changes maintenance fork of Symfony 1.4. That's something we might take on if L'Express prefers to build Symfony 1.5 (which of course is a fine goal, it's just not our goal).

**Comment by GromNaN** [comment]
Thanks all for your interest in our symfony1 fork.
As you might guess, we upgraded all our old applications to this version and that gave us a lot of benefices compared to the time we spent fixing some forms.
Indeed, we did it because we need to maintain and enhance these applications for some time.

**Comment by sgloe** [comment]
There is of course a huge interest of a full-bc! The forms are holding us back, too, switching to 1.5.

**Comment by pbowyer** [comment]
@boutell That would be fantastic. Happy to contribute, but don't have the manpower to do such a fork myself.

**Comment by stevelacey** [comment]
@GromNaN is there nothing of value in 1.5 that would be worth back-porting into a 1.4 branch? e.g. backwards compatible improvements? Seems like a duplication of effort for @boultell to fire up another fork, when this repo could represent both use-cases. i.e. switch to a 1.4.x tag if you need full compat.

**Comment by boutell** [comment]
I would certainly start by cherry-picking commits rather than reinventing
the wheel.

On Sat, Jan 24, 2015 at 8:39 AM, Steve Lacey <notifications@github.com>
wrote:

> @GromNaN <https://github.com/GromNaN> is there nothing of value in 1.5
> that would be worth back-porting into a 1.4 branch? e.g. backwards
> compatible improvements? Seems like a duplication of effort for @boultell
> to fire up another fork, when this repo could represent both use-cases.
> i.e. switch to a 1.4.x tag if you need full compat.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LExpress/symfony1/issues/54#issuecomment-71316930>.
>



-- 


*THOMAS BOUTELL, *DEV & OPS
P'UNK AVENUE | (215) 755-1330  |  punkave.com

**Comment by boutell** [comment]
It does look like the great majority of commits here are for new features.

On Sat, Jan 24, 2015 at 8:52 AM, Tom Boutell <tom@punkave.com> wrote:

> I would certainly start by cherry-picking commits rather than reinventing
> the wheel.
>
> On Sat, Jan 24, 2015 at 8:39 AM, Steve Lacey <notifications@github.com>
> wrote:
>
>> @GromNaN <https://github.com/GromNaN> is there nothing of value in 1.5
>> that would be worth back-porting into a 1.4 branch? e.g. backwards
>> compatible improvements? Seems like a duplication of effort for @boultell
>> to fire up another fork, when this repo could represent both use-cases.
>> i.e. switch to a 1.4.x tag if you need full compat.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/LExpress/symfony1/issues/54#issuecomment-71316930>.
>>
>
>
>
> --
>
>
> *THOMAS BOUTELL, *DEV & OPS
> P'UNK AVENUE | (215) 755-1330  |  punkave.com
>



-- 


*THOMAS BOUTELL, *DEV & OPS
P'UNK AVENUE | (215) 755-1330  |  punkave.com

**Comment by stevelacey** [comment]
Yeah, I'm sure everyone's onboard that that's the way to go, I am just suggesting that if it can exist here, in this repo, and the distinction between the patched 1.4 and 1.5 branches made clear, it should.

**Comment by boutell** [comment]
That could be a good thing, yes. Our problem is that we need to move our
Ubuntu 10.04 clients to a newer LTS by April 1st, and in the process it
would be irresponsible not to address other components of their stack that
are not being maintained or shortly will not be

Which means we need to move them to PHP 5.4+, the drak bc fork of Doctrine
1.2, and a maintained fork of Symfony 1.4 very promptly.

So if we need to in order to hit our timeline, we'll do our own fork, but
I'm up for merging with any project that has compatible goals.


On Sat, Jan 24, 2015 at 9:14 AM, Steve Lacey <notifications@github.com>
wrote:

> Yeah, I'm sure everyone's onboard that that's the way to go, I am just
> suggesting that if it can exist here, in this repo, and the distinction
> between the patched 1.4 and 1.5 branches made clear, it should.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LExpress/symfony1/issues/54#issuecomment-71318636>.
>



-- 


*THOMAS BOUTELL, *DEV & OPS
P'UNK AVENUE | (215) 755-1330  |  punkave.com

**Comment by boutell** [comment]
We've released a strict-bc fork here:

https://github.com/punkave/symfony1

Based on cherrypicking the PHP 5.4/5.5 compatibility commits from the L'Express fork.

We also documented a procedure to use the backwards-compatible drak fork of Doctrine 1.2 with it, and to install and update both of them without using composer or upsetting an existing svn workflow.

We've credited the L'Express fork with bringing the 5.4/5.5 compatibility stuff into the world, of course.

Looking forward to a mutually beneficial relationship between these two projects.
